[JIT] correct torch.jit.Attribute docs

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #74653

torch.jit.Attribute needs to be used in subclasses of
torch.jit.ScriptModule, not torch.nn.Module

davidberard98 committed:
[JIT] correct torch.jit.Attribute docs

torch.jit.Attribute needs to be used in subclasses of
torch.jit.ScriptModule, not torch.nn.Module

[ghstack-poisoned]

davidberard98 committed:
Update on "[JIT] correct torch.jit.Attribute docs"


torch.jit.Attribute needs to be used in subclasses of
torch.jit.ScriptModule, not torch.nn.Module

[ghstack-poisoned]

sharvil commented:
I think it's worth pointing out that this using `torch.jit.Attribute` is no longer the preferred way of declaring attributes on a `Module`. The preferred way is described here: https://pytorch.org/docs/stable/jit.html#attributes.

davidberard98 committed:
Update on "[JIT] correct torch.jit.Attribute docs"


torch.jit.Attribute needs to be used in subclasses of
torch.jit.ScriptModule, not torch.nn.Module

[ghstack-poisoned]

davidberard98 committed:
Update on "[JIT] correct torch.jit.Attribute docs"


torch.jit.Attribute needs to be used in subclasses of
torch.jit.ScriptModule, not torch.nn.Module

[ghstack-poisoned]

eellison reviewed (approved):
LGTM, thanks!

davidberard98 commented:
@pytorchbot merge this please
